packages/flutter_tools/lib/src/macos/build_macos.dart

```diff
@@ -181,11 +181,25 @@ Future<void> _writeCodeSizeAnalysis(BuildInfo buildInfo, SizeAnalyzer? sizeAnaly
   if (buildInfo.codeSizeDirectory == null || sizeAnalyzer == null) {
     return;
   }
-  final String arch = DarwinArch.x86_64.name;
-  final File aotSnapshot = globals.fs.directory(buildInfo.codeSizeDirectory)
-    .childFile('snapshot.$arch.json');
-  final File precompilerTrace = globals.fs.directory(buildInfo.codeSizeDirectory)
-    .childFile('trace.$arch.json');
+  final File? aotSnapshot = DarwinArch.values.map<File?>((DarwinArch arch) {
```

Comment:
Theoretically which architecture we use here is dependent on the order in which the enum values are declared, which could change. However, when considering the tiny likelihood of this happening and the tiny impact it would probably have on users, this is fine.

Reply:
right, this is also assuming they're comparing size-analysis snapshots across flutter/dart sdk upgrades.